cmd/fleetctl/debug.go

```diff
@@ -316,3 +322,144 @@ func debugArchiveCommand() *cli.Command {
 		},
 	}
 }
+
+func debugConnectionCommand() *cli.Command {
+	// TODO: do we want to allow setting this via a flag?
+	const timeoutPerCheck = 10 * time.Second
+
+	return &cli.Command{
+		Name:      "connection",
+		ArgsUsage: "[<address>]",
+		Usage:     "Investigate the cause of a connection failure to the Fleet server.",
+		Description: `Run a number of checks to debug a connection failure to the Fleet
+server.
+
+If <address> is provided, this is the address that is investigated,
+otherwise the address of the provided context is used, with
+the default context used if none is explicitly specified.`,
+		Flags: []cli.Flag{
+			configFlag(),
+			contextFlag(),
+			debugFlag(),
+			fleetCertificateFlag(),
+		},
+		Action: func(c *cli.Context) error {
+			var addr string
+			if narg := c.NArg(); narg > 0 {
+				if narg > 1 {
+					return errors.New("too many arguments")
+				}
+				addr = c.Args().First()
+			}
+
+			// TODO: not correct: clientFromCLI requires a login token, which
+			// can only be obtained once the connection works. The context cannot apply
+			// for this command, use a provided address (as mentioned in the issue), and
+			// build the fleet client just before the API endpoint check.
+			_ = addr
```

Comment:
Maybe `unauthenticatedClientFromCLI`?

Reply:
Thanks, I missed that option. I ended up refactoring the clientFromCLI/unauthenticatedClientFromCLI a bit so I could get the context config separately (while reusing the existing validations in place regarding context existing and ensuring the config file exists), and override the address with the explicitly-provided one if present before creating the client (I still need the context config for the TLS insecure flag and root CA if configured).